packages/peregrine/lib/context/cart.js

```diff
@@ -1,41 +1,58 @@
-import React, { createContext, useContext, useMemo } from 'react';
+import React, { createContext, useContext, useEffect, useMemo } from 'react';
+import { useLazyQuery } from '@apollo/react-hooks';
 import { connect } from 'react-redux';
 
 import actions from '../store/actions/cart/actions';
 import * as asyncActions from '../store/actions/cart/asyncActions';
 import bindActionCreators from '../util/bindActionCreators';
 
+// TODO Where would we store this?
+import GET_CART_DETAILS_QUERY from '../../../venia-ui/lib/queries/getCartDetails.graphql';
+
 const CartContext = createContext();
 
-const isCartEmpty = cart =>
-    !cart || !cart.details.items || cart.details.items.length === 0;
+const isCartEmpty = cart => !cart || !cart.items || cart.items.length === 0;
 
 const getTotalQuantity = items =>
     items.reduce((total, item) => total + item.quantity, 0);
 
 const CartContextProvider = props => {
     const { actions, asyncActions, cartState, children } = props;
 
+    const { cartId } = cartState;
+    const [fetchCartDetails, { data = {} }] = useLazyQuery(
+        GET_CART_DETAILS_QUERY
+    );
+
+    useEffect(() => {
+        if (cartId) {
+            fetchCartDetails({
```

Comment:
Instead of fetching here we could, like I did in cart trigger, fetch just data we need in the components.